XXX-url.md

```diff
@@ -0,0 +1,170 @@
+| Title  | Implement WHATWG URL Spec   |
+|--------|-----------------------------|
+| Author | @jasnell                    |
+| Status | DRAFT                       |
+| Date   | 2016-06-30T09:00:00-07:00   |
+
+## Description
+
+The WHATWG URL Standard specifies updated syntax, parsing and serialization of
+URLs as currently implemented by the main Web Browsers. The existing Node.js
+`url` module parsing and serialization implementation currently does not support
+the URL standard and fails to pass 160 of the WHATWG URL parsing tests.
+
+This proposal is to implement the WHATWG URL Standard by modifying the existing
+`url` module to provide an implementation of the `URL` object and associated
+APIs. Doing so improves the robustness of URL parsing, provides consistency
+with browser js code, and can eventually allow the reduction of node.js specific
+APIs.
+
+Initially, the implementation would be introduced as an undocumented
+experimental feature exposed via a new `URL` property in the `url` module.
+
+Because the existing `require('url')` module remains, there should be no
+backwards compatibility concerns. Once a decision is made to provide official,
+documented support for the new URL implementation, a decision would be made
+about whether to switch the internal uses of `require('url').parse()` to the
+new `URL` implementation.
```

Comment:
https://github.com/nodejs/node/issues/4677
On Jul 3, 2016 6:41 PM, "Chris Dickinson" <notifications@github.com> wrote:

> In XXX-url.md
> <https://github.com/nodejs/node-eps/pull/28#discussion_r69401408>:
>
> > +the URL standard and fails to pass 160 of the WHATWG URL parsing tests.
> > +
> > +This proposal is to implement the WHATWG URL Standard by modifying the existing
> > +`url` module to provide an implementation of the `URL` object and associated
> > +APIs. Doing so improves the robustness of URL parsing, provides consistency
> > +with browser js code, and can eventually allow the reduction of node.js specific
> > +APIs.
> > +
> > +Initially, the implementation would be introduced as an undocumented
> > +experimental feature exposed via a new `URL` property in the `url` module.
> > +
> > +Because the existing `require('url')` module remains, there should be no
> > +backwards compatibility concerns. Once a decision is made to provide official,
> > +documented support for the new URL implementation, a decision would be made
> > +about whether to switch the internal uses of `require('url').parse()` to the
> > +new `URL` implementation.
>
> @Trott <https://github.com/Trott>, @jokeyrhyme
> <https://github.com/jokeyrhyme>: There are multiple levels of deprecation
> — there's "admonition against use in the docs" deprecation, and there's
> "actively emits warnings when used" deprecation. Deprecating
> (-as-in-warning) and removing the old url parser is probably unrealistic
> given the number of users it would affect.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/nodejs/node-eps/pull/28/files/a0a892210e9fa20f684247dd58e838a9cd333df1#r69401408>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAa2eeyQ_OfDFjsVyOxcKHk17rk8sBwtks5qSGTjgaJpZM4IrtaC>
> .
>